src/transport/CryptoContext.h

```diff
@@ -152,20 +158,13 @@ class DLL_EXPORT CryptoContext
     bool IsResponder() const { return mKeyAvailable && mSessionRole == SessionRole::kResponder; }
 
 private:
-    typedef uint8_t CryptoKey[Crypto::kAES_CCM128_Key_Length];
-
-    enum KeyUsage
-    {
-        kI2RKey                  = 0,
-        kR2IKey                  = 1,
-        kAttestationChallengeKey = 2,
-        kNumCryptoKeys           = 3
-    };
-
     SessionRole mSessionRole;
 
     bool mKeyAvailable;
-    CryptoKey mKeys[KeyUsage::kNumCryptoKeys];
+    Crypto::Aes128KeyHandle mEncryptionKey;
+    Crypto::Aes128KeyHandle mDecryptionKey;
+    Crypto::Aes128KeyByteArray mAttestationChallenge;
```

Comment:
Suggest creating a new class for SharedSecretByteArray<size_t N> to handle the IRK and the attestation challenge.